src/components/Header/NetworkSelector.tsx

```diff
@@ -425,7 +328,14 @@ export default function NetworkSelector() {
             </FlyoutHeader>
             {NETWORK_SELECTOR_CHAINS.map((chainId: SupportedChainId) =>
               isChainAllowed(connector, chainId) ? (
-                <Row onSelectChain={onSelectChain} targetChain={chainId} key={chainId} />
+                <Row
+                  onSelectChain={(targetChainId: SupportedChainId) => {
+                    selectChain(targetChainId)
+                    closeModal()
```

Comment:
Should this wait until the selectChain callback resolves?

Reply:
yeah i can change that